pkgs/applications/science/logic/tlaplus/toolbox.nix

```diff
@@ -0,0 +1,80 @@
+{ lib, fetchzip, makeWrapper, makeDesktopItem, stdenv, jre, swt, gtk, libXtst, glib
+, ... }:
+
+let
+  version = "1.5.6";
+  arch = "x86_64";
+
+  desktopItem = makeDesktopItem rec {
+    name = "TLA+Toolbox";
+    exec = "tla-toolbox";
+    icon = "tla-toolbox";
+    comment = "IDE for TLA+";
+    desktopName = name;
+    genericName = comment;
+    categories = "Application;Development";
+    extraEntries = ''
+      StartupWMClass=TLA+ Toolbox
+    '';
+  };
+
+
+in stdenv.mkDerivation {
+  name = "tla-toolbox-${version}";
+  src = fetchzip {
+    url = "https://tla.msr-inria.inria.fr/tlatoolbox/products/TLAToolbox-${version}-linux.gtk.${arch}.zip";
+    sha256 = "1h63mcbrkf4jcg6qncpqffdi0x665z0wlfdq43d67p411xcqmbw9";
+  };
+
+  buildInputs = [ makeWrapper  ];
+
+  phases = [ "installPhase" ];
+
+  installPhase = ''
+    mkdir -pv "$out/bin"
+    cp -rv "$src" "$out/toolbox"
+    chmod +w "$out/toolbox/toolbox"
+
+    patchelf \
+      --set-interpreter $(cat $NIX_CC/nix-support/dynamic-linker) \
+      "$out/toolbox/toolbox"
+
+    echo "cd $(echo $out/toolbox); ./toolbox -data ~/.tla-toolbox \"$@\"" \
+      > "$out/bin/tla-toolbox"
+
+    chmod +x $out/bin/tla-toolbox
+
+    wrapProgram "$out/bin/tla-toolbox" \
```

Comment:
if you use `makeWrapper`, you don't need the `echo` above. It also has a `--run` parameter.